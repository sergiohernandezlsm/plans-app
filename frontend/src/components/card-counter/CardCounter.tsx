import React from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import CardPlan from '../card';
import { CardTypes } from '../../types';

interface Props {
  cards: CardTypes[];
}

const CardCounter: React.FC<Props> = ({ cards }) => {
  return (
    <>
      <Row className="justify-content-center">
        {cards.map((card, index) => {
          return (
            <Col xs={12} key={`key-${index}`}>
              <CardPlan
                card={card}
              />
            </Col>
          )
        })}
      </Row>
    </>
  )
};

export default CardCounter;
