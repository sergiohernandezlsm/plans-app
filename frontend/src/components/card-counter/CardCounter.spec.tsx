import { shallow } from 'enzyme';
import CardCounter, { Props } from './CardCounter';

const props: Props = {
  subscriptionsData: [{ id: 1 }, { id: 2 }],
  selectionHandler: () => { },
}

describe('CardCounter Component', () => {
  it('renders <CardCounter /> component', () => {
    const wrapper = shallow(<CardCounter {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
