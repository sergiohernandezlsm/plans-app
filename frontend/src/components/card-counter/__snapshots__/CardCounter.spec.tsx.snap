// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCounter Component renders <CardCounter /> component 1`] = `ShallowWrapper {}`;
